### PR TITLE
Track `GenericArgs` with Closure to support `to_rustc`

### PR DESCRIPTION
**Pull request by ranjitjhala**

### CODE REVIEW

File: crates/flux-middle/src/rty/mod.rs [hunk 1]
```diff
@@ -1280,7 +1285,7 @@ pub enum BaseTy {
     Tuple(List<Ty>),
     Array(Ty, Const),
     Never,
-    Closure(DefId, /* upvar_tys */ List<Ty>),
+    Closure(DefId, /* upvar_tys */ List<Ty>, GenericArgs),
```
Comment (nilehmann) [inline crates/flux-middle/src/rty/mod.rs:1288]: I thought we were going to store a `flux_rustc_bridge::ty::GenericArgs` so we don't have to worry about the refinements, i.e., we store the "shape".

Comment (ranjitjhala) [reply crates/flux-middle/src/rty/mod.rs:1288]: We need to convert back to `rustc` -- is there a `to_rustc` for the `flux_rustc_bridge::ty::GenericArgs` ?

Comment (ranjitjhala) [reply crates/flux-middle/src/rty/mod.rs:1288]: oh wait there is...

**Review by nilehmann — APPROVED** [review]
:shipit:

**Pull request merged**